miniworld/wrappers.py

```diff
@@ -40,3 +42,27 @@ def observation(self, obs):
         obs = 0.30 * obs[:, :, 0] + 0.59 * obs[:, :, 1] + 0.11 * obs[:, :, 2]
 
         return np.expand_dims(obs, axis=2)
+
+
+class StochasticActionWrapper(gym.ActionWrapper):
+    """
+    Add stochasticity to the actions
+
+    If a random action is provided, it is returned with probability `1 - prob`.
+    Else, a random action is sampled from the action space.
+    """
+
+    def __init__(self, env=None, prob=0.9, random_action=None):
+        super().__init__(env)
+        self.prob = prob
+        self.random_action = random_action
+
+    def action(self, action):
+        """ """
+        if np.random.uniform() < self.prob:
+            return action
+        else:
+            if self.random_action is None:
+                return random.randint(0, 6)
```

Comment:
This should use `self.np_random_randint` such that `seed` will affect the randomness to be deterministic

Reply:
Made the suggested change by using `self.np_random.integers`